sklearn/decomposition/_dict_learning.py

```diff
@@ -1017,15 +1024,15 @@ def __init__(self, dictionary, transform_algorithm='omp',
                  transform_n_nonzero_coefs=None, transform_alpha=None,
                  split_sign=False, n_jobs=None, positive_code=False,
                  transform_max_iter=1000):
-        self._set_sparse_coding_params(dictionary.shape[0],
```

Comment:
We do not define `n_components` anymore. Some this be somewhere?

Reply:
Uhm right, that's why I add a property which I removed.

I can reintroduce it and deprecate it because this is actually useless and was never documented.